cucumber-junit-platform-engine/src/main/java/io/cucumber/junit/platform/engine/FeaturesPropertyResolver.java

```diff
@@ -0,0 +1,107 @@
+import org.junit.platform.engine.DiscoveryFilter;
+import org.junit.platform.engine.DiscoverySelector;
+import org.junit.platform.engine.EngineDiscoveryRequest;
+
+import java.util.List;
+import java.util.Set;
+
+import static io.cucumber.junit.platform.engine.Constants.FEATURES_PROPERTY_NAME;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Decorator to support resolving the
+ * {@value io.cucumber.junit.platform.engine.Constants#FEATURES_PROPERTY_NAME}
+ * property.
+ * <p>
+ * The JUnit Platform provides various discovery selectors to select feature
+ * files. Unfortunately, these do not yet receive support from IDEs, Maven or
+ * Gradle. Resolving this property allows uses to target a single feature,
+ * scenario or example from the commandline.
+ * <p>
+ * This class decorates the {@link DiscoverySelectorResolver}. When the features
+ * property is provided it replaces the discovery request.
+ * <p>
+ * Note: This effectively causes Cucumber to ignore any requests from the JUnit
+ * Platform. So features will be discovered even when none are expected to be.
+ */
+class FeaturesPropertyResolver {
+
+    private static final Logger log = LoggerFactory.getLogger(FeaturesPropertyResolver.class);
+    private static boolean warnedWhenCucumberFeaturesPropertyIsUsed = false;
+
+    private final DiscoverySelectorResolver delegate;
+
+    FeaturesPropertyResolver(DiscoverySelectorResolver delegate) {
+        this.delegate = delegate;
+    }
+
+    void resolveSelectors(EngineDiscoveryRequest request, CucumberEngineDescriptor engineDescriptor) {
+        ConfigurationParameters configuration = request.getConfigurationParameters();
+        CucumberEngineOptions options = new CucumberEngineOptions(configuration);
+        Set<FeatureWithLinesSelector> selectors = options.featuresWithLines();
+
+        if (selectors.isEmpty()) {
+            delegate.resolveSelectors(request, engineDescriptor);
+            return;
+        }
+
+        warnWhenCucumberFeaturesPropertyIsUsed();
+        EngineDiscoveryRequest replacement = new FeaturesPropertyDiscoveryRequest(request, selectors);
+        delegate.resolveSelectors(replacement, engineDescriptor);
+    }
+
+    private static void warnWhenCucumberFeaturesPropertyIsUsed() {
+        if (warnedWhenCucumberFeaturesPropertyIsUsed) {
+            return;
+        }
+        warnedWhenCucumberFeaturesPropertyIsUsed = true;
+        log.warn(
+            () -> "Discovering tests using the " + FEATURES_PROPERTY_NAME + " property. Other discovery " +
+                    "selectors are ignored!\n" +
+                    "\n" +
+                    "This is a work around for the limited JUnit 5 support in Maven and Gradle. " +
+                    "Please request/upvote/sponsor/ect better support for JUnit 5 discovery selectors. " +
+                    "For details see: https://github.com/cucumber/cucumber-jvm/pull/2498\n" +
```

Comment:
This should also be interesting for you (if/when it gets merged):
https://github.com/gradle/gradle/pull/29535